src/test/java/com/wikia/webdriver/common/templates/NewTestTemplateCore.java

```diff
@@ -125,41 +122,22 @@ protected void setWindowSize(int width, int height, WebDriver desiredDriver) {
     desiredDriver.manage().window().setSize(dimension);
   }
 
-  protected void setBrowserUserAgent(String userAgent) {
-    NewDriverProvider.setBrowserUserAgent(config.getBrowser(), userAgent);
-  }
-
-  protected void runProxyServerIfNeeded(Method method) {
-    boolean isGeoEdgeSet = false;
-    boolean isNetworkTrafficDumpSet = false;
-    String countryCode = null;
-
-    if (method.getAnnotation(GeoEdgeProxy.class) != null) {
-      isGeoEdgeSet = true;
-      countryCode = method.getAnnotation(GeoEdgeProxy.class).country();
-    }
-
-    if (method.getAnnotation(NetworkTrafficDump.class) != null) {
-      isNetworkTrafficDumpSet = true;
-    }
-
-    if (isGeoEdgeSet || isNetworkTrafficDumpSet) {
-      isProxyServerRunning = true;
-      networkTrafficIntereceptor = new NetworkTrafficInterceptor();
-      networkTrafficIntereceptor.startSeleniumProxyServer();
-    } else {
-      return;
-    }
-
-    if (isGeoEdgeSet) {
-      GeoEdgeUtils geoEdgeUtils = new GeoEdgeUtils(config.getCredentialsFilePath());
-      String credentialsBase64 = "Basic " + geoEdgeUtils.createBaseFromCredentials();
-      String ip = geoEdgeUtils.getIPForCountry(countryCode);
-      networkTrafficIntereceptor.setProxyServer(ip);
-      networkTrafficIntereceptor.changeHeader("Proxy-Authorization", credentialsBase64);
+  public void runProxyServer(String countryCode) {
+    isProxyServerRunning = true;
+    networkTrafficIntereceptor = new NetworkTrafficInterceptor();
+    networkTrafficIntereceptor.startSeleniumProxyServer();
+    if (countryCode != null) {
```

Comment:
here is how you can deal with all this null checks = http://en.wikipedia.org/wiki/Null_Object_pattern